web_api/web_app.py

```diff
@@ -0,0 +1,123 @@
+from flask import Flask, request, jsonify
+from datetime import datetime
+import uuid
+import os
+import json
+
+from werkzeug.utils import secure_filename
+
+app = Flask(__name__)
+app.config['UPLOAD_FOLDER'] = os.path.join(__file__, "..", "..", "uploads")
+app.config['OUTPUTS_FOLDER'] = os.path.join(__file__, "..", "..", "outputs")
+
+
+class RequestStatus:
+    DONE = "done"
+    PENDING = 'pending'
+    NOT_FOUND = "not found"
+
+
+@app.route('/upload', methods=['POST'])
+def upload():
+    """
+    Handle file upload.
+
+    This endpoint receives a file in a POST request and saves it in the uploads folder.
+    It returns a JSON object with a unique identifier (UID) for the uploaded file.
+
+    :return: JSON response with the UID of the upload.
+    """
+    if not os.path.isdir(app.config['UPLOAD_FOLDER']):
+        os.mkdir(app.config['UPLOAD_FOLDER'])
+
+    if 'file' not in request.files:
+        return jsonify({'error': 'No file provided.'}), 400
+
+    file = request.files['file']
+    uid = str(uuid.uuid4())
+
+    original_base_filename, original_file_type = _get_lecture_name_and_type(secure_filename(file.filename))
+    timestamp = datetime.now().strftime('%Y-%m-%d-%H-%M-%S')
+
+    new_filename = f"{original_base_filename}_{timestamp}_{uid}.{original_file_type}"
+    file.save(os.path.join(app.config['UPLOAD_FOLDER'], new_filename))
+    return jsonify({'uid': uid})
+
+
+def _get_explanation_from_json_file(file_path: str) -> list[str]:
```

Comment:
it would be better if you move these function to another file, to keep only endpoints and allow better reading

Reply:
done